### PR TITLE
br: Add annotations for warmup pod

### PR DESCRIPTION
**Pull request by wxiaomou**
### What problem does this PR solve?

The warmup pod does not have annotation on Restore CR, this PR carry the annotations to the warmup pod.

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/backup/restore/restore_manager.go [hunk 1]
```diff
@@ -1160,7 +1160,13 @@ func (rm *restoreManager) makeSyncWarmUpJob(r *v1alpha1.Restore, tc *v1alpha1.Ti
 	}
 	resourceRequirements := getWarmUpResourceRequirements(tc)
 
+	jobAnnotations := r.Annotations
+	podAnnotations := jobAnnotations
+
 	warmUpPod := &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: podAnnotations,
+		},
```
Comment (BornChanger) [inline pkg/backup/restore/restore_manager.go:1169]:
```suggestion
	warmUpPod := &corev1.PodTemplateSpec{
		ObjectMeta: metav1.ObjectMeta{
			Annotations: r.Annotations,
		},
```

File: cmd/http-service/pbgen/api/service.pb.go
Comment (WangLe1321) [inline cmd/http-service/pbgen/api/service.pb.go:1]: Please revert the change of this file, or we can't cherry-pick this PR to release-1.5

File: cmd/http-service/pbgen/api/service_grpc.pb.go
Comment (WangLe1321) [inline cmd/http-service/pbgen/api/service_grpc.pb.go:1]: Please revert the change of this file, or we can't cherry-pick this PR to release-1.5

File: pkg/backup/restore/restore_manager.go
Comment (WangLe1321) [inline pkg/backup/restore/restore_manager.go:1]: you should also modify the codes here https://github.com/pingcap/tidb-operator/blob/master/pkg/backup/restore/restore_manager.go#L1252

File: cmd/http-service/pbgen/api/service.pb.go [hunk 2]
```diff
@@ -7,13 +7,14 @@
 package api
 
 import (
+	reflect "reflect"
```
Comment (BornChanger) [inline cmd/http-service/pbgen/api/service.pb.go:10]: unnecessary change.

File: cmd/http-service/pbgen/api/service_grpc.pb.go [hunk 3]
```diff
@@ -8,6 +8,7 @@ package api
 
 import (
 	context "context"
+
```
Comment (BornChanger) [inline cmd/http-service/pbgen/api/service_grpc.pb.go:11]: unnecessary change.

File: cmd/http-service/pbgen/api/service.pb.go (hunk 2 above)
Comment (wxiaomou) [reply cmd/http-service/pbgen/api/service.pb.go]: done

File: cmd/http-service/pbgen/api/service_grpc.pb.go (hunk 3 above)
Comment (wxiaomou) [reply cmd/http-service/pbgen/api/service_grpc.pb.go]: done

File: pkg/backup/restore/restore_manager.go
Comment (wxiaomou) [reply pkg/backup/restore/restore_manager.go:1]: @WangLe1321 Thanks, done

File: cmd/http-service/pbgen/api/service_grpc.pb.go
Comment (wxiaomou) [reply cmd/http-service/pbgen/api/service_grpc.pb.go:1]: done

File: cmd/http-service/pbgen/api/service.pb.go
Comment (wxiaomou) [reply cmd/http-service/pbgen/api/service.pb.go:1]: done

File: pkg/backup/restore/restore_manager.go [hunk 4]
```diff
@@ -1160,7 +1160,16 @@ func (rm *restoreManager) makeSyncWarmUpJob(r *v1alpha1.Restore, tc *v1alpha1.Ti
 	}
 	resourceRequirements := getWarmUpResourceRequirements(tc)
 
+	jobAnnotations := r.Annotations
+	podAnnotations := jobAnnotations
+	jobLabels := r.Labels
```
Comment (WangLe1321) [inline pkg/backup/restore/restore_manager.go:1165]: we have `jobLabel` definition below, so the `jobLabels` here is a little confused. I suggest to remove definition of `jobAnnotations` and `jobLabels`.

Comment (wxiaomou) [reply pkg/backup/restore/restore_manager.go]: @WangLe1321 done. Thanks

**Comment by wxiaomou** [comment]
@WangLe1321  addressed comments, could you take another look?

**Review by WangLe1321 — APPROVED** [review]
